app/connectors/ContentHttpReads.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import cats.implicits._
+import connectors.httpParser.JourneyStateParser.pagerDutyError
+import play.api.libs.json.Reads
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+class ContentHttpReads[A: Reads] extends HttpReads[ContentResponse[Option[A]]] {
+  override def read(method: String, url: String, response: HttpResponse): ContentResponse[Option[A]] =
+    if (isSuccess(response.status)) response.json.validate[A].asOpt.asRight else Left(pagerDutyError(response))
```

Comment:
Just thinking, could we simplify it and not return our right wrapped in an Option? 

```scala
class ContentHttpReads[A: Reads] extends HttpReads[ContentResponse[A]] {
  override def read(method: String, url: String, response: HttpResponse): ContentResponse[A] =
    if (isSuccess(response.status)) {
      response.json.validate[A].fold(_ => HttpError.`someParsingError`.asLeft, _.asRight)
    }
    else {
      Left(pagerDutyError(response))
    }
}

```

Reply:
good point. Let's not mask the error in option but return left of error when it fails